src/Symfony/Component/DependencyInjection/Tests/Fixtures/php/services9_compiled.php

```diff
@@ -99,11 +99,7 @@ protected function getBarService()
      */
     protected function getBazService()
     {
-        $this->services['baz'] = $instance = new \Baz();
-
-        $instance->setFoo($this->get('foo_with_inline'));
-
-        return $instance;
+        return $this->services['baz'] = new \Baz();
```

Comment:
This was a case where the circular reference was *working*, as the service can be instantiated and registered in `` $this->services`` before the point triggering the call to the other service, meaning it will receive the instance.
Forbidding a case which was working previously is a BC break. So this is not acceptable, especially for a patch release